Create simple Shader framework

pearpanda opened the pull request:
Set up CMake to copy `res/` folder to the build directory when building;

Created simple Shader framework which supports:
- Compiling and linking shader components (vertex, fragment)
- Compilation and linking error logging (with `spdlog`)
- Binding and unbinding
- Uniform setting for common types;

Added a utility function to read files into C++ strings

Commit:
Add utility functions to read file into string

Commit:
Add Shader abstraction

Implemented a Shader class with the following features:
- Compiling and linking shader components (vertex, fragment)
- Compilation and linking error logging (with `spdlog`)
- Binding and unbinding
- Uniform setting for common types

Commit:
Changed main.cpp: Rotating triangle example

Commit:
Reformat files

Commit:
Add `const` to Shader::get_uniform_location

Commit:
Correct typo

luka-j reviewed (approved):
Meni ovo deluje kul, imam malo mixed feelings oko Shader_set.cpp (može i samo da bude na dnu Shader.cpp, malo izdvojen?), mada kontam što si to uradio, to su samo utility metode koje nećemo gledati. 
I ovaj read_file.cpp mi malo štrči, ali ako bude potrebe za još nekom srodnom metodom možemo da je dodamo tu i preimenujemo fajl kad dođe vreme (npr. ako nekad zatreba write_file, ja bih je dodao tu i preimenovao fajl da se zove io_utils; a možda i ne bude write nego nešto što sad ne vidimo).

Commit:
Update `const` specifiers for Shader

Commit:
Add Shader uniform setter, fix debug OpenGL call

Integer uniforms can now be set using the `Shader` class, which comes
in handy when working with textures.

The previous version had a bug where `glGetProgramiv` was called with
the `GL_COMPILE_STATUS` parameter instead of `GL_LINK_STATUS`.
